scripts/partition_manager/partition_manager.rst

```diff
@@ -0,0 +1,408 @@
+
+Usage
+=====
+When using Partition Manager in your application, you should be aware of the functionality it adds.
+
+C code
+------
+When Partition Manager is enabled, all source files are compiled with the define ``USE_PARTITION_MANAGER`` set to 1.
+If you use this define in your code, the preprocessor can choose what code to include depending on whether Partition Manager is being used.
+
+.. code-block:: C
+
+   #if USE_PARTITION_MANAGER
+   #include <pm_config.h>
+   #define NON_SECURE_APP_ADDRESS PM_APP_ADDRESS
+   #else
+   ...
+
+HEX files
+---------
+Partition Manager may implicitly or explicitly assign a HEX file to a partition.
+
+Image partitions are implicitly assigned the compiled HEX file, i.e. the HEX file that is generated when building the corresponding image.
+Phony partitions are implicitly assigned the result of merging the HEX files that are assigned to the underlying partitions.
+Placeholder partitions are not implicitly assigned a HEX file.
+
+To explicitly assign a HEX file to a partition, set the global properties *partition_name*\ _PM_HEX_FILE and *partition_name*\ _PM_TARGET in CMake, where *partition_name* is the name of the partition.
+*partition_name*\ _PM_TARGET specifies the build target that generates the HEX file specified in *partition_name*\ _PM_HEX_FILE.
+
+See the following example, which assigns a cryptographically signed HEX file built by the ``sign_target`` build target to the root application:
+
+
+.. code-block:: cmake
+
+   set_property(
+     GLOBAL PROPERTY
+     app_PM_HEX_FILE # Must match "*_PM_HEX_FILE"
+     ${PROJECT_BINARY_DIR}/signed.hex
+   )
+
+   set_property(
+     GLOBAL PROPERTY
+     app_PM_TARGET # Must match "*_PM_TARGET"
+     sign_target
+   )
+
+
+As output, Partition Manager creates a HEX file called :file:`merged.hex`, which is flashed to the board when calling ``ninja flash``.
+When creating :file:`merged.hex`, all assigned HEX files are included in the merge operation.
+If the HEX files overlap, the conflict is resolved as follows:
+
+   * HEX files assigned to phony partitions overwrite HEX files assigned to their underlying partitions.
+   * HEX files assigned to larger partitions overwrite HEX files assigned to smaller partitions.
+   * Explicitly assigned HEX files overwrite implicitly assigned HEX files.
+
+This means that you can overwrite a partition's HEX file by wrapping that partition in another partition, and assigning a HEX file to the new partition.
+
+ROM report
+----------
+When using the Partition Manager, run ``ninja rom_report`` to see the addresses and sizes of flash partitions.
+
+.. _pm_cmake_usage:
+
+CMake
+-----
+The CMake variables from Partition Manager are typically used through generator expressions, because these variables are made available at the end of the CMake configure stage.
```

Comment:
I have no idea what generator expressions are. I'm fine leaving as is though if we think readers know. ;-)

Reply:
Alternatively one can point to https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html as it is something not all are familiar with.